pfioh/mount_dir.py

```diff
@@ -0,0 +1,179 @@
+"""
+Handle MountDir file storage
+"""
+
+import shutil
+import datetime
+import zipfile
+import zlib
+import gzip
+import base64
+import os
+import ast
+from   io                   import BytesIO
+from   io                   import StringIO
+from   pfioh                import StoreHandler
+from   pfioh                import base64_process, zip_process, zipdir
+try:
+    from ._colors import Colors
+except:
+    from _colors import Colors
+
+class MountDir(StoreHandler):
+
+    def __init__(self,*args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.qprint('MountDir initialized')
+        
+
+    def storeData(self, **kwargs):
+        """
+        Stores the file/directory at the specified location
+        """
+
+
+        for k,v in kwargs.items():
+            if k == 'file_name': fileName  = v
+            if k == 'file_content': fileContent = v
+            if k == 'Path': unpackPath= v
+            if k == 'is_zip': b_zip= v
+            if k == 'd_ret': d_ret= v
+
+        with open(fileName, 'wb') as fh:
+            try:
+                fh.write(fileContent)
+            except Exception as err:
+                self.qprint(err)
+
+        fileSize = self.getSize(fileName)
+
+        if b_zip:
+            try:
+                d_fio = zip_process(
+                    action= 'unzip',
+                    path= unpackPath,
+                    payloadFile = fileName
+                )
+                d_ret['unzip']  = d_fio
+                d_ret['status'] = d_fio['status']
+                d_ret['msg']    = d_fio['msg']
+            finally:
```

Comment:
I think this is at the wrong place.  It needs to correspond to the open above.

Reply:
I didn't understand this. If I move it, to correspond with open, it will delete the zip file before unpacking.

Reply:
Move the try up to the open.  And put the finally just above the if b_zip.